src/server/zset_family.cc

```diff
@@ -2166,26 +2180,19 @@ void ZSetFamily::ZCount(CmdArgList args, const CommandContext& cmd_cntx) {
   }
 }
 
-void ZSetFamily::ZDiff(CmdArgList args, const CommandContext& cmd_cntx) {
-  vector<vector<ScoredMap>> maps(shard_set->size());
-  auto cb = [&](Transaction* t, EngineShard* shard) {
-    maps[shard->shard_id()] = OpFetch(shard, t);
-    return OpStatus::OK;
-  };
-
-  cmd_cntx.tx->ScheduleSingleHop(std::move(cb));
+/* Calculate difference between key set and all other sets. */
+vector<ScoredMemberView> ZDiffOp(vector<OpResult<vector<ScoredMap>>>& maps, ScoredMap &result, ShardId key_sid) {
```

Comment:
our style guide requires that mutable argument will be passed by pointer.
So in that case you can pass `map` as temporary `.. && map` object or pass it by value and move into it from a caller.

the result, should be `ScoreMap* result`. Also the output variables should be the last ones, so `ShardId key_sid` should preferrably be the first argument